### ISSUE TITLE
pytest logging is awkward

### ISSUE THREAD

**Issue by benclifford** [Parsl/parsl#300]
The current test suite seems to conflate logs of various test runs, because they all run in the same python process, and logging is a global feature, not constrained to any one test or DFK.

This makes it hard to look at a test rundir afterwards and understand what has happened.

**Comment by annawoodard** [comment]
Agreed. The fact that logging is a global feature is a problem with nose or any other testing suite that I know of. The purpose of the unique rundirs was to allow for running the tests with multiple processes-- we had some tests relying on parsing the logs-- I've tried to remove them but I think there may still be some remaining. Maybe we should name the rundirs with the test name?

**Comment by benclifford** [comment]
added the monitoring tag to this -- monitoring is moving in a direction where it might have much richer scope information around things happening, beyond "this logger is in the DFK source file", and while this issue is not phrased as a monitoring issue, there is potential to implement this separation as an incidental part of that monitoring work.
